core/services/workflows/engine.go

```diff
@@ -245,9 +258,23 @@ func (e *Engine) loop(ctx context.Context) {
 	}
 }
 
+func generateExecutionID(workflowID, eventID string) (string, error) {
+	s := sha256.New()
+	_, err := s.Write([]byte(workflowID))
+	if err != nil {
+		return "", err
+	}
+
+	_, err = s.Write([]byte(eventID))
+	if err != nil {
+		return "", err
+	}
+
+	return base64.StdEncoding.EncodeToString(s.Sum(nil)), nil
```

Comment:
We need to guarantee that IDs are 32 bytes long (forwarder assumption)

Reply:
Hmmm... we can use the raw bytes as the id throughout, but that seems messy (for one it's difficult to copy and paste into logs). What are you suggesting I do instead? It should be possible for whatever is downstream to take this ID and convert it into 32byte if they need it.

Reply:
One option is to use raw 32 bytes but just be careful to always log as hex so it's greppable.

Reply:
On second thoughts, maybe let's keep using hex-encoded strings and only decode to raw bytes at the last step.